demos/common/src/main/java/dk/alexandra/fresco/demo/cli/CmdLineProtocolSuite.java

```diff
@@ -119,27 +141,121 @@ private Properties getProperties(Properties properties) {
     return properties;
   }
 
-  private SpdzResourcePool createSpdzResourcePool(Properties properties) {
-    String strat = properties.getProperty("spdz.preprocessingStrategy");
+  private SpdzResourcePool createSpdzResourcePool(Properties properties,
+      Supplier<Network> networkSupplier) {
+
+    String strat = properties.getProperty("spdz.preprocessingStrategy", "DUMMY");
     final PreprocessingStrategy strategy = PreprocessingStrategy.valueOf(strat);
+
+    final int modBitLength = Integer.parseInt(properties.getProperty("spdz.modBitLength", "128"));
+    final BigInteger modulus = ModulusFinder.findSuitableModulus(modBitLength);
+    final BigIntegerFieldDefinition definition = new BigIntegerFieldDefinition(modulus);
     SpdzDataSupplier supplier = null;
-    if (strategy == PreprocessingStrategy.DUMMY) {
-      BigInteger modulus = ModulusFinder.findSuitableModulus(512);
-      supplier = new SpdzDummyDataSupplier(myId, noOfPlayers,
-          new BigIntegerFieldDefinition(modulus), modulus);
-    }
-    if (strategy == PreprocessingStrategy.STATIC) {
-      int noOfThreadsUsed = 1;
-      String storageName =
-          SpdzStorageDataSupplier.STORAGE_NAME_PREFIX + noOfThreadsUsed + "_" + myId + "_" + 0
-              + "_";
-      supplier = new SpdzStorageDataSupplier(
-          new FilebasedStreamedStorageImpl(new InMemoryStorage()), storageName, noOfPlayers);
+
+    switch (strategy) {
```

Comment:
switch is very hard to test - that could explain some of the loss of coverage.

Reply:
That, and the exp-pipe generation is not tested. Hence the "equals demo", which is now moved to the demos-common in order to get 100% codecov in that project.